exonum-java-binding-core/src/main/java/com/exonum/binding/crypto/Ed25519CryptoFunction.java

```diff
@@ -38,50 +35,58 @@
 
   INSTANCE;
 
-  private static final int CRYPTO_SIGN_ED25519_SEEDBYTES = 64;
+  private final LazySodiumJava lazySodium = new LazySodiumJava(new SodiumJava());
 
   @Override
   public KeyPair generateKeyPair(byte[] seed) {
-    checkArgument(hasLength(seed, CRYPTO_SIGN_ED25519_SEEDBYTES),
-        "Seed byte array has invalid size (%s), must be 64", seed.length);
+    checkArgument(hasLength(seed, SEED_BYTES),
+        "Seed byte array has invalid size (%s), must be %s", seed.length, SEED_BYTES);
+
+    byte[] publicKey = lazySodium.randomBytesBuf(PUBLIC_KEY_BYTES);
+    byte[] privateKey = lazySodium.randomBytesBuf(PRIVATE_KEY_BYTES);
 
-    byte[] privateKey = zeros(CRYPTO_SIGN_ED25519_SECRETKEYBYTES);
-    byte[] publicKey = zeros(CRYPTO_SIGN_ED25519_PUBLICKEYBYTES);
-    if (!checkReturnValueSuccess(
-        sodium().crypto_sign_ed25519_seed_keypair(publicKey, privateKey, seed))) {
+    if (!lazySodium.cryptoSignSeedKeypair(publicKey, privateKey, seed)) {
       throw new RuntimeException("Failed to generate a key pair");
     }
     return KeyPair.createKeyPairNoCopy(privateKey, publicKey);
   }
 
   @Override
   public KeyPair generateKeyPair() {
-    return generateKeyPair(new Random().randomBytes(CRYPTO_SIGN_ED25519_SEEDBYTES));
+    byte[] publicKey = lazySodium.randomBytesBuf(PUBLIC_KEY_BYTES);
+    byte[] privateKey = lazySodium.randomBytesBuf(PRIVATE_KEY_BYTES);
+
+    if (!lazySodium.cryptoSignKeypair(publicKey, privateKey)) {
+      throw new RuntimeException("Failed to generate a key pair");
+    }
+    return KeyPair.createKeyPairNoCopy(privateKey, publicKey);
   }
 
   @Override
   public byte[] signMessage(byte[] message, PrivateKey privateKey) {
-    byte[] signedMessage = Util.prependZeros(CRYPTO_SIGN_ED25519_BYTES, message);
-    LongLongByReference bufferLen = new LongLongByReference(0);
-    sodium()
-        .crypto_sign_ed25519(
-            signedMessage, bufferLen, message, message.length, privateKey.toBytesNoCopy());
-    byte[] signature = slice(signedMessage, 0, CRYPTO_SIGN_ED25519_BYTES);
-    return signature;
+    byte[] signedMessage = lazySodium.randomBytesBuf(SIGNATURE_BYTES + message.length);
+
+    if (!lazySodium
```

Comment:
That's a very poor API, accepting length of the signature and message :disappointed: :
* (a) when it's known from the Java array size;
* (b) as a long array!
* (c) `null` being OK?

Would you please fire an issue?